BMS Workspace-git/bms-1227-fw/Phantom_drivers/soc/soc.c

```diff
@@ -31,7 +31,7 @@ float battCapacity = 0; // Max chemical capacity for cell TODO: Make static afte
 static TickType_t xLastWakeTime = 0;
 float lastCurrentValue = 0; // TODO: Make static after unit-testing
 
-State_t state = CHARGE; // TODO: Make static after unit-testing 
+SOCState_t SOCState = CHARGE; // TODO: Make static after unit-testing 
 bms_data* BMSDataPtr; // TODO: Make this extern and initialize BMSDataPtr when BMS is initialized
```

Comment:
I'd rather you have the pointer as an input to the init function and set it that way

Reply:
CHARGE was the wrong thing to initialize it to. I actually mean to initialize it to INIT, and then the state machine knows to call the SOCinit() function based on this state. Does that flow make sense?

Reply:
Changed to be initialized to INIT.
This way, the state machine knows to call the SOCinit() function based on this state the first time SOCprocess() is called. Does that flow make sense?